### PR TITLE
Handle RefreshRequiredEventType

### PR DESCRIPTION
**Pull request by peterbryntesson**
fix: handling refresh required event by calling ConditionRefresh() on the subscription

### CODE REVIEW

**Comment by peterbryntesson** [comment]
> The ConditionRefresh logic you added is replicated, right? Can we extract it to a separate method and reuse that?

Maybe. We could put the code in SubscriptionWrapper - but here in the code we only have access to the raw subscription. I decided to just duplicate the code would be better - but I can change it.

**Review by simonjaeger — APPROVED** [review]
LGTM

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs [hunk 1]
```diff
@@ -1460,6 +1460,31 @@ public class MonitoredItemWrapper {
                         _pendingAlarmsSnapshotTimer.Start();
                         return;
                     }
+                    else if (eventType == ObjectTypeIds.RefreshRequiredEventType) {
+                        var noErrorFound = true;
+
+                        // issue a condition refresh to make sure we are in a correct state
+                        _logger.Information("Now issuing ConditionRefresh for subscription " +
```
Comment (mregen) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs:1467]: has there been an ask to support conditionrefresh2 as well?

**Review by mregen — APPROVED** [review]

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs (hunk 1 above)
Comment (peterbryntesson) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs:1467]: No, not that I have heard about.

**Pull request merged**